src/Language/PureScript/Errors.hs

```diff
@@ -432,9 +443,13 @@ onTypesInErrorMessageM f (ErrorMessage hints simple) = ErrorMessage <$> traverse
   gSimple (OverlappingInstances cl ts insts) = OverlappingInstances cl <$> traverse f ts <*> pure insts
   gSimple (PossiblyInfiniteInstance cl ts) = PossiblyInfiniteInstance cl <$> traverse f ts
   gSimple (CannotDerive cl ts) = CannotDerive cl <$> traverse f ts
+  gSimple (CannotDeriveNullaryTypeClassInstance strat cl) = pure $ CannotDeriveNullaryTypeClassInstance strat cl
   gSimple (InvalidNewtypeInstance cl ts) = InvalidNewtypeInstance cl <$> traverse f ts
-  gSimple (MissingNewtypeSuperclassInstance cl1 cl2 ts) = MissingNewtypeSuperclassInstance cl1 cl2 <$> traverse f ts
-  gSimple (UnverifiableSuperclassInstance cl1 cl2 ts) = UnverifiableSuperclassInstance cl1 cl2 <$> traverse f ts
+  gSimple (InvalidViaType cl ts viaTy) = InvalidViaType cl <$> traverse f ts <*> f viaTy
+  gSimple (InvalidViaKind cl ts viaTy viaKind expectedKind) = InvalidViaKind cl <$> traverse f ts <*> f viaTy <*> f viaKind <*> f expectedKind
```

Comment:
Should `f` be called on kinds here? There’s no case for the `KindsDoNotUnify` error for instance so maybe it shouldn’t 🤔

Reply:
If there are cases that are missing it's probably an oversight.

Reply:
The following `SimpleErrorMessage` constructors hold `SourceType`s but aren’t matched:

+ `InfiniteKind`
+ `EscapedSkolem`
+ `KindsDoNotUnify`
+ `MissingClassMember`
+ `UserDefinedWarning`
+ `QuantificationCheckFailureInType`
+ `UnsupportedTypeInKind`

Should I add cases for them in this PR? Matching every constructor explicitly would be verbose but could help to prevent this in the future.